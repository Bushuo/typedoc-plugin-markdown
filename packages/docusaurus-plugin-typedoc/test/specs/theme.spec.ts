import * as Handlebars from 'handlebars';
import { TestApp } from '../../../typedoc-plugin-markdown/test/test-app';

describe(`Theme:`, () => {
  let testApp: TestApp;
  let reflectionTemplate: Handlebars.TemplateDelegate;

  beforeAll(async () => {
    testApp = new TestApp(['theme.ts']);
    await testApp.bootstrap();
    TestApp.stubPartials(['index', 'comment', 'member.signature', 'members']);
    TestApp.stubHelpers(['breadcrumbs', 'hierarchy']);
    reflectionTemplate = TestApp.getTemplate('reflection');
  });

  describe(`(header)`, () => {
    test(`should compile with module without subtitle`, () => {
      expect(
        TestApp.compileTemplate(reflectionTemplate, {
          model: testApp.project.children[0],
          project: testApp.project,
        }),
      ).toMatchSnapshot();
    });

    test(`should compile with module with subtitle`, () => {
      expect(
        TestApp.compileTemplate(reflectionTemplate, {
          model: testApp.findReflection('InterfaceItemA').children[0],
          project: testApp.project,
        }),
      ).toMatchSnapshot();
    });
  });
});
