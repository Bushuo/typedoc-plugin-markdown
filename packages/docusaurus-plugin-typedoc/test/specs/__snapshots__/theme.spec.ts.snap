// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Theme: (header) should compile with module with subtitle 1`] = `
"[helper: breadcrumbs]

# Property: prop

[InterfaceItemA](interfaceitema.md).prop

[helper: toc]

[partial: members]
"
`;

exports[`Theme: (header) should compile with module without subtitle 1`] = `
"[helper: breadcrumbs]

# Namespace: namespaceA

[helper: toc]

[partial: members]
"
`;
